File: MvvmSample/app/src/main/java/com/abben/mvvmsample/base/BaseAndroidViewModel.java
package com.abben.mvvmsample.base;

import android.app.Application;
import androidx.annotation.NonNull;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;
import io.reactivex.disposables.CompositeDisposable;

/**
 * @author abben
 * on 2018/12/23
 */
public abstract class BaseAndroidViewModel extends AndroidViewModel {
    protected CompositeDisposable compositeDisposable;
    protected MutableLiveData<String> toastLiveData = new MutableLiveData<>();

    public BaseAndroidViewModel(@NonNull Application application) {
        super(application);
        compositeDisposable = new CompositeDisposable();
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        compositeDisposable.clear();
    }

    public MutableLiveData<String> getToastLiveData() {
        return toastLiveData;
    }
}

File: MvvmSampleJetpack/app/src/main/java/com/infun/mvvmsamplejetpack/base/BaseAndroidViewModel.kt
package com.infun.mvvmsamplejetpack.base

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import io.reactivex.disposables.CompositeDisposable

/**
 *  @author abben
 *  on 2019/1/17
 */
class BaseAndroidViewModel(private val baseApplication: Application) : AndroidViewModel(baseApplication) {
    var compositeDisposable: CompositeDisposable? = null
    val toastLiveData: MutableLiveData<String> = MutableLiveData()

    init {
        compositeDisposable = CompositeDisposable()
    }

    override fun onCleared() {
        super.onCleared()
        compositeDisposable?.clear()
    }
}